diff --git a/lemonbar.c b/lemonbar.c
index de4cf73..6a2f7cd 100644
--- a/lemonbar.c
+++ b/lemonbar.c
@@ -26,6 +26,8 @@
 #define max(a,b) ((a) > (b) ? (a) : (b))
 #define min(a,b) ((a) < (b) ? (a) : (b))
 
+#define MAX_FONT_COUNT 5
+
 typedef struct font_t {
     xcb_font_t ptr;
     int descent, height, width;
@@ -98,6 +100,9 @@ static int font_index = -1;
 static uint32_t attrs = 0;
 static bool dock = false;
 static bool topbar = true;
+static int offsets_y[MAX_FONT_COUNT];
+static int offset_y_count = 0;
+static int offset_y_index = 0;
 static int bw = -1, bh = -1, bx = 0, by = 0;
 static int bu = 1; // Underline height
 static rgba_t fgc, bgc, ugc;
@@ -260,7 +265,7 @@ int xft_char_width (uint16_t ch, font_t *cur_font) {
 int draw_char_xft(monitor_t *mon, font_t *cur_font, int x, int align, uint16_t ch) {
     int ch_width = xft_char_width(ch, cur_font);
     x = shift(mon, x, align, ch_width);
-    int y = bh / 2 + cur_font->height / 2- cur_font->descent;
+    int y = bh / 2 + cur_font->height / 2- cur_font->descent + offsets_y[offset_y_index];
     XftDrawString16(xft_draw, &sel_fg, cur_font->xft_font, x, y, &ch, 1);
     draw_lines(mon, x, ch_width);
     return ch_width;
@@ -530,8 +535,10 @@ select_drawable_font (const uint16_t c)
     // If the end is reached without finding an appropriate font, return NULL.
     // If the font can draw the character, return it.
     for (int i = 0; i < font_count; i++) {
-        if (font_has_glyph(font_list[i], c))
+        if (font_has_glyph(font_list[i], c)) {
+        	offset_y_index = i;
             return font_list[i];
+           }
     }
     return NULL;
 }
@@ -1583,7 +1590,14 @@ main (int argc, char **argv)
                         "\t-F Set foreground color in #AARRGGBB\n", argv[0]);
                 exit (EXIT_SUCCESS);
             case 'g': (void)parse_geometry_string(optarg, geom_v); break;
-            case 'o': (void)parse_output_string(optarg); break;
+            case 'o': offsets_y[offset_y_count] = strtol(optarg, NULL, 10);
+            	if (offset_y_count == 0) {
+            		for (int i=1; i<MAX_FONT_COUNT; i++) {
+            			offsets_y[i] = offsets_y[0];
+            		}
+            	}
+            	++offset_y_count;
+            	break;
             case 'p': permanent = true; break;
             case 'n': wm_name = xstrdup(optarg); break;
             case 'b': topbar = false; break;
